rust/operator/src/container.rs

```diff
@@ -720,14 +737,32 @@ impl ContainerConfig {
         &self,
         merged_config: &(dyn MergedConfig + Send + 'static),
     ) -> Option<ResourceRequirements> {
-        // Only the Hadoop main containers will get resources
         match self {
+            // name node and journal node resources
             ContainerConfig::Hdfs { role, .. } if role != &HdfsRole::DataNode => {
                 merged_config.resources().map(|c| c.into())
             }
+            // init containers inherit their respective main (app) container resources
+            ContainerConfig::FormatNameNodes { .. } | ContainerConfig::FormatZooKeeper { .. } => {
+                merged_config.resources().map(|c| c.into())
+            }
+            // name node side car zk failover (see https://docs.google.com/spreadsheets/d/1iWX1g4HaY3sFN9846BYd8kXZDU6FQkwSPmILsWMClE0/edit#gid=379007403)
```

Comment:
I *think* I prefer not link to an internal google docs from an public code
```suggestion
            // Namenode zk failover sidecar
```

Reply:
Yeah thought about that as well. Will remove.